nimbus/src/client/mod.rs

```diff
@@ -0,0 +1,40 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+mod fs_client;
+mod http_client;
+use crate::error::{Error, Result};
+use crate::Experiment;
+use fs_client::FileSystemClient;
+use http_client::Client;
+use url::Url;
+
+use crate::ExperimentConfig;
+
+pub(crate) fn create_client(
+    collection_name: &str,
+    config: ExperimentConfig,
+) -> Result<Box<dyn SettingsClient>> {
+    // XXX - double-parsing the URL here if it's not a file:// URL - ideally
+    // config would already be holding a Url and we wouldn't parse here at all.
+    let url = Url::parse(&config.server_url)?;
+    Ok(if url.scheme() == "file" {
+        // collection_name is ignored for the file-system - we could insist
```

Comment:
In fact, it's not obvious to me why `collection_name` is passed as a separate argument rather than as an additional property of the `config`.

Reply:
This comment about `collection_name` is out of date now and should be removed.